x-pack/plugins/observability_solution/inventory/public/components/entity_icon/index.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { AGENT_NAME, CLOUD_PROVIDER } from '@kbn/observability-shared-plugin/common';
+import { type CloudProvider, CloudProviderIcon, AgentIcon } from '@kbn/custom-icons';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import type { AgentName } from '@kbn/elastic-agent-utils';
+import { ENTITY_TYPE } from '../../../common/es_fields/entities';
+import type { Entity } from '../../../common/entities';
+
+interface EntityIconProps {
+  entity: Entity;
+}
+
+export function EntityIcon({ entity }: EntityIconProps) {
+  const entityType = entity[ENTITY_TYPE];
+  const emptyIcon = (
```

Comment:
We don't need to return an empty div here.  Both Cloud and Agent will return the default icon.

Reply:
I am using the `emptyIcon` in case `entityType` is not `host`, `contanier` or `service`.

Reply:
When fetching the data we filter only these entity types: 
- https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/inventory/server/routes/entities/get_latest_entities.ts#L34 
- https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/inventory/server/routes/entities/query_helper.ts#L12 

but even if we want to handle this the UI in case it might change,  it would be better to use EuiIcon with an empty icon instead of hardcoding an empty div. wdyt?